src/content/app/entity-viewer/state/bookmarks/entityViewerBookmarksSlice.ts

```diff
@@ -85,38 +109,14 @@ export const updatePreviouslyViewedEntities = createAsyncThunk(
   }
 );
 
-export const updatePreviouslyViewedEntity = createAsyncThunk(
-  'entity-viewer/updatePreviouslyViewedEntity',
-  (
-    params: {
-      genomeId: string;
-      previouslyViewedEntityId: string;
-      fragment: Partial<PreviouslyViewedEntity>;
-    },
-    thunkAPI
-  ) => {
-    const { genomeId, previouslyViewedEntityId, fragment } = params;
-    const state = thunkAPI.getState() as RootState;
-    const updatedEntities = getPreviouslyViewedEntities(state, genomeId).map(
-      (entity) => {
-        if (entity.entity_id === previouslyViewedEntityId) {
-          return merge({}, entity, fragment);
-        }
-        return entity;
-      }
+export const loadPreviouslyViewedEntities =
+  (): ThunkAction<void, any, null, Action<string>> => (dispatch) => {
+    dispatch(
+      bookmarksSlice.actions.setInitialPreviouslyViewedEntities(
+        entityViewerBookmarksStorageService.getPreviouslyViewedEntities()
+      )
     );
```

Comment:
You can just return whatever you get from `entityViewerBookmarksStorageService.getPreviouslyViewedEntities()` from `loadPreviouslyViewedEntities` (without dispatching anything) and add a handler for `loadPreviouslyViewedEntities.fulfilled` in `extraReducers` in the slice

Reply:
That's done